lib/child_process.js

```diff
@@ -245,39 +243,29 @@ exports.execFile = function(file /*, args, options, callback*/) {
   }
 
   if (child.stdout) {
-    if (encoding)
-      child.stdout.setEncoding(encoding);
-
     child.stdout.addListener('data', function(chunk) {
-      stdoutLen += chunk.length;
+      stdoutLen += (chunk.byteLength || chunk.length);
```

Comment:
Wait, what? Doesn't this bring us back to the original problem if `chunk` is a string?